src/jefferson.boilerplate-middleware.test.js

```diff
@@ -0,0 +1,195 @@
+"use strict";
+var chai = require("chai"),
+    express = require("express"),
+    request = require("supertest"),
+    jefferson = require("./jefferson"),
+    middleware = require("../test/middleware");
+let expect = chai.expect;
+
+let checkRequest = (invocation, expected) => {
+    return new Promise((resolve, reject) => {
+        invocation.end((err, res) => {
+            if (err) {
+                return reject(err);
+            }
+            expect(res.text).to.equal(expected);
+            resolve();
+        });
+    });
+};
+
+describe("Jefferson Pre/Post-Middleware", () => {
+    it("can describe middleware that's invoked before all chains", () => {
+        let conf = {
+            pre: {
+                all: [
+                    middleware.append(0),
+                    middleware.append(1)
+                ]
+            },
+            routes: {
+                "getItem": {
+                    method: "GET",
+                    path: "/test",
+                    middleware: [
+                        middleware.append(2),
+                        middleware.append(3),
+                        middleware.sendResult
+                    ]
+                }
+            }
+        };
+        let app = express();
+        jefferson(app, conf);
+        return checkRequest(request(app).get("/test"), "0123");
+    });
+
+    it("can describe middleware that's invoked before mutable and immutable chains", () => {
+        let conf = {
+            post: {
+                all: [middleware.sendResult]
+            },
+            pre: {
+                all: [middleware.append("0")],
+                safe: [middleware.append("I")],
+                unsafe: [middleware.append("M")],
+                method: {
+                    get: [middleware.append(1)],
+                    put: [middleware.append(2)],
+                    post: [middleware.append(3)],
+                    delete: [middleware.append(4)],
+                    options: [middleware.append(5)]
+                }
+            },
+            routes: {
```

Comment:
Ok, nm, you just weren't covering all those other methods in the tests - that's fine. I would cover anything that seems likely to be used, but again, I'm not sure *patch* is even going to be common anyway.